addons/website_event/models/website.py

```diff
@@ -23,3 +24,13 @@ def _search_get_details(self, search_type, order, options):
         if search_type in ['events', 'all']:
             result.append(self.env['event.event']._search_get_detail(self, order, options))
         return result
+
+    @api.model
+    def new_page(self, name=False, add_menu=False, template='website.default_page', ispage=True, namespace=None, page_values=None, menu_values=None, sections_arch=None):
+        is_event = False
+        if template.startswith('website_event.') and template != 'website_event.template_intro':
+            add_menu, ispage, is_event = False, False, True
+        event_page = super().new_page(name=name, add_menu=add_menu, template=template, ispage=ispage, namespace=namespace, page_values=page_values, menu_values=menu_values, sections_arch=sections_arch)
+        if is_event and request:
+            event_page['url'] = f"/event/{request.params.get('event_id')}/page" + event_page['url']
```

Comment:
Otherwise event page will have the website url not the controller handling events [Here](https://github.com/odoo/odoo/blob/0d755e71278db1e03f30478729590cf177407c48/addons/website/models/website.py#L573-L576)

Reply:
Perhaps, we should add a comment on how the URL will look after this operation.